src/libYARP_dev/src/modules/RGBDSensorClient/RGBDSensorClient.cpp

```diff
@@ -0,0 +1,464 @@
+// -*- mode:C++; tab-width:4; c-basic-offset:4; indent-tabs-mode:nil -*-
+
+/*
+ * Copyright (C) 2016 iCub Facility - Istituto Italiano di Tecnologia
+ * Author: Alberto Cardellino <alberto.cardellino@iit.it>
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ *
+ */
+
+#include "RGBDSensorClient.h"
+#include <yarp/os/Log.h>
+#include <yarp/os/LogStream.h>
+
+using namespace yarp::dev;
+using namespace yarp::os;
+using namespace yarp::sig;
+
+
+// needed for the driver factory.
+yarp::dev::DriverCreator *createRGBDSensorClient() {
+    return new DriverCreatorOf<yarp::dev::RGBDSensorClient>("RGBDSensorClient",
+        "RGBDSensorClient",
+        "yarp::dev::RGBDSensorClient");
+}
+
+RGBDSensorClient::RGBDSensorClient()
```

Comment:
you could be using initializers instead of assigments here, i.e.
```c++
RGBDSensorClient::RGBDSensorClient() : 
        hDim(640),
        vDim(480),
        watchdog(-1),
[...]
{
}
```